src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -762,18 +758,14 @@ private static GcvScope getScope(
                 continue;
             }
 
-            traversedComponents.add(dependent);
-            stack.addAll(dependent.getSelected().getDependents());
+            stack.addAll(dependent.getFrom().getDependents());
         }
 
         GcvScope scope = discoveredScopes.stream()
                 .min(GCV_SCOPE_COMPARATOR)
                 .orElseThrow(() -> new RuntimeException("Couldn't determine scope for dependency: " + component));
 
         scopeCache.put(component, scope);
-        for (ResolvedDependencyResult traversedComponent : traversedComponents) {
-            scopeCache.put(traversedComponent.getSelected(), scope);
-        }
```

Comment:
Coudln't the removal of this trigger a pathological N^2 case if getScope is called on the leaves first and so on going up the graph?
Even if that's not necessarily the case, wouldn't want to rely on implicit gradle behaviour as we know they change things like this all the time.